docs/guide.rst

```diff
@@ -31,109 +31,134 @@ The `WGPU <https://github.com/pygfx/wgpu-py>`_ library provides the low level AP
 
-To render your scene to the canvas, you need a renderer. A renderer
-needs a target to render to (which can be a canvas or a texture).
-And finally, you need a camera to define the point of view to render the scene from.
+    # run the application
+    run()
 
-.. code-block:: python
+.. image:: _static/guide_static_cube.png
 
-    # A renderer is associated with a canvas (or a texture) that it renders to
-    renderer = gfx.renderers.WgpuRenderer(canvas)
+Nice, we rendered our first scene! Next-up we can look at a full example that
+takes this one step further and animates the cube.
 
-    # A camera defines the viewpoint and projection
-    camera = gfx.PerspectiveCamera(70, 16 / 9)
+.. _full_example:
 
-    ...
+Animations
+----------
 
-    # The actual rendering
-    renderer.render(scene, camera)
+As promised in the previous section, here is a full example of how to use pygfx.
+It adds a little bit of flare to the hello world example by rotating the cube a
+bit during the draw call. This allows us to create a simple animation::
 
-
-Putting it together
--------------------
-
-If you run this, you should see a rotating yellow cube.
-
-.. code-block:: python
+    from wgpu.gui.auto import WgpuCanvas, run
 
     import pygfx as gfx
 
-    from PySide6 import QtWidgets
-    from wgpu.gui.qt import WgpuCanvas
-
+    scene = gfx.Scene()
 
-    app = QtWidgets.QApplication([])
+    # create a camera to view the scene
+    camera = gfx.PerspectiveCamera(70, 16 / 9)
+    camera.position.z = 400
 
-    # Create a canvas and a renderer
-    canvas = WgpuCanvas()
-    renderer = gfx.renderers.WgpuRenderer(canvas)
+    # add some light so that the camera can see
+    scene.add(gfx.AmbientLight())
+    scene.add(gfx.DirectionalLight())
 
-    # Populate a scene with a cube
-    scene = gfx.Scene()
-    geometry = gfx.BoxGeometry(200, 200, 200)
+    # Populate the scene
+    geometry = gfx.box_geometry(200, 200, 200)
     material = gfx.MeshPhongMaterial(color=(1, 1, 0, 1))
     cube = gfx.Mesh(geometry, material)
     scene.add(cube)
+    
+    # Create a canvas and a renderer
+    canvas = WgpuCanvas()
+    renderer = gfx.renderers.WgpuRenderer(canvas)
 
-    camera = gfx.PerspectiveCamera(70, 16 / 9)
-    camera.position.z = 400
 
-    def animate():
-        rot = gfx.linalg.Quaternion().set_from_euler(gfx.linalg.Euler(0.005, 0.01))
+    def draw_function():
+        # custom logic to rotate the cube
+        rot = gfx.linalg.Quaternion().set_from_euler(
+            gfx.linalg.Euler(0.005, 0.01)
+        )
         cube.rotation.multiply(rot)
 
+        # update the internal buffer
         renderer.render(scene, camera)
-        canvas.request_draw()
 
-    canvas.request_draw(animate)
-    app.exec_()
+        # Note: the handover to the canvas is implicit
+
+        # schedule the next draw call to show an animation
+        # Note: without arguments it will use the previous draw_function
+        renderer.request_draw()
```

Comment:
Worth noting (here, in the PR) that `renderer.request_draw()` is a wrapper for `canvas.request_draw()`. We use the renderer's version because then we don't need a reference to the canvas, leading to simpler code. So I don't think we need to mention it in the guide.

Reply:
Hm, so as a user of pygfx would I be expected to not interact with the canvas, or would I need to configure it somehow?

My angle is that, if I am expected to interact with a canvas, then I would wonder what the value of the `renderer.request_draw` function is. However, if I am not expected to directly interact with a canvas, then I would wonder why we need to introduce it as one of the first things in the guide. 

Following that thought, I would also wonder if it would make sense to just have a default value for the canvas in `gfx.renderers.WgpuRenderer()` (my suggestion is the `auto` canvas) that a user can overwrite if they need to. That way we only need to bring in a canvas.

Reply:
Sorry if I only added confusion :)  I just wanted to mention it as "background". I don't think we should mention it in the guide, because in pygfx you typically don't need the canvas.

Reply:
> because in pygfx you typically don't need the canvas.

In that case I would vote for leaving the canvas out of the guide completely. Perhaps we can combine this with a feature PR that makes the auto canvas the default canvas for a renderer?

Reply:
I would be against hiding it completely because it's still a useful concept. We could discuss this further, but let's do that in another issue/pr.

Reply:
This is now resolved by #378 and #380.